src/main/java/com/appland/appmap/AgentCommandLineLoader.java

```diff
@@ -0,0 +1,63 @@
+package com.appland.appmap;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableList;
+import org.gradle.api.GradleException;
+import org.gradle.api.Named;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.Internal;
+import org.gradle.process.CommandLineArgumentProvider;
+import org.gradle.util.RelativePathUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AgentCommandLineLoader implements CommandLineArgumentProvider, Named {
+    private static final Logger LOGGER = Logging.getLogger(CommandLineArgumentProvider.class);
+
+    private final AppmapPluginExtension appmap;
+
+    public AgentCommandLineLoader(AppmapPluginExtension appmap) {
+        this.appmap = appmap;
+    }
+
+    @Override
+    public Iterable<String> asArguments() {
+        return getAsJvmArg();
+    }
+
+    @SuppressWarnings("NullableProblems")
+    @Internal
+    @Override
+    public String getName() {
+        return "appmapAgent";
+    }
+
+    @Internal
+    public List<String> getAsJvmArg() {
+        if (!appmap.isConfigFileValid()) {
+            appmap.setSkip(true);
+            throw new GradleException("Configuration file must exist and be readable: "
+                    + appmap.getConfigFile().get().getAsFile().getPath());
+        }
+        if (appmap.shouldSkip()) {
+            LOGGER.warn("Appmap task was executed but but is disable, skip property set to " + appmap.shouldSkip());
+            return new ArrayList<>();
+        } else {
+            String builder = "-javaagent:" +
+                    RelativePathUtil.relativePath(
+                            appmap.project.getProjectDir(), appmap.getAgentConf().getSingleFile()
+                    );
+            List<String> argumentLn = ImmutableList.of(
+                    builder,
+                    "-Dappmap.config.file=" + appmap.getConfigFile().get().toString(),
+                    "-Dappmap.output.directory=" + appmap.getOutputDirectory().get().toString(),
+                    "-Dappmap.event.valueSize=" + appmap.getEventValueSize(),
+                    "-Dappmap.debug=" + appmap.getDebug()
+            );
```

Comment:
Check the Maven logic for handling the debug flags:
https://github.com/applandinc/appmap-maven-plugin/blob/master/src/main/java/com/appland/appmap/AppMapAgentMojo.java#L101-L111

In summary: presence of any non-null, non-empty value will introduce `-Dappmap.debug`, `-Dappmap.debug.file=${debugFile}`. The value of [`debug`](https://github.com/applandinc/appmap-gradle-plugin/blob/12377fb0947b3f544b78fc124e1428b9081cc3a3/src/main/java/com/appland/appmap/AppmapPluginExtension.java#L28) will then be split by comma (the Maven plugin also allows for whitespace but doesn't document it), and appends `-Dappmap.debug.hooks`, `-Dappmap.debug.locals`, `-Dappmap.debug.http` as appropriate if the last token is included in the list.

Reply:
I added the changes requested, but let me do some testing first, first time Monday I will address this.

Reply:
@dustinbyrne @ptrdvrk finally got to test this properly, there is an issue with the gradle plugin that should be solved once is properly mad public, for now if you want to play with it to you can only do it with groovy gradle scripts (build.gradle) but not with kotlin DSL bild scripts (build.gradle.kts)

For testing I used:
https://github.com/spring-guides/tut-spring-boot-kotlin

but had to change the build.gradle.kts to build.gradle because of the aforementioned, you can use it if you like:


```
buildscript {
	repositories {
		mavenLocal()
	}
	dependencies {
		classpath "com.appland:appmap-gradle-plugin:1.0.3"
	}
}

plugins {
	id "org.springframework.boot" version "2.4.4"
	id "io.spring.dependency-management" version "1.0.11.RELEASE"
	id "org.jetbrains.kotlin.jvm" version "1.4.32"
	id "org.jetbrains.kotlin.plugin.spring" version "1.4.32"
	id "org.jetbrains.kotlin.plugin.allopen" version "1.4.32"
	id "org.jetbrains.kotlin.plugin.jpa" version "1.4.32"
	id "org.jetbrains.kotlin.kapt" version "1.4.32"
}

group = "com.example"
version = "0.0.1-SNAPSHOT"
java.sourceCompatibility = JavaVersion.VERSION_1_8

repositories {
	mavenCentral()
}

dependencies {
	implementation "org.springframework.boot:spring-boot-starter-data-jpa"
	implementation "org.springframework.boot:spring-boot-starter-mustache"
	implementation "org.springframework.boot:spring-boot-starter-web"
	implementation "com.fasterxml.jackson.module:jackson-module-kotlin"
	implementation "org.jetbrains.kotlin:kotlin-reflect"
	implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
	runtimeOnly "com.h2database:h2"
	runtimeOnly "org.springframework.boot:spring-boot-devtools"
	kapt "org.springframework.boot:spring-boot-configuration-processor"
	testImplementation('org.springframework.boot:spring-boot-starter-test') {
		exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
		exclude module: 'mockito-core'
	}
	testImplementation "org.junit.jupiter:junit-jupiter-api"
	testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine"
	testImplementation "com.ninja-squad:springmockk:3.0.1"
	test.useJUnitPlatform()
}

compileKotlin {
	kotlinOptions.freeCompilerArgs =  ['-Xjsr305=strict']
	kotlinOptions.jvmTarget = "1.8"
}

allOpen {
	annotation("javax.persistence.Entity")
	annotation("javax.persistence.Embeddable")
	annotation("javax.persistence.MappedSuperclass")
}

apply plugin: 'com.appland.appmap'

appmap {
	configFile = file("$projectDir/appmap.yml")
	outputDirectory = file("$buildDir/appmap3")
	skip = false
	debug = "info"
	eventValueSize = 1024
}
```


Once you validate this, please merge


Thanks!